src/core/mac/sub_mac_csl_receiver.cpp

```diff
@@ -191,36 +200,52 @@ void SubMac::HandleCslTimer(void)
     }
     else
     {
-        if (RadioSupportsReceiveTiming())
-        {
-            mCslTimer.FireAt(mCslSampleTime - timeAhead + periodUs);
-            timeAhead -= kCslReceiveTimeAhead;
-            winStart = mCslSampleTime.GetValue() - timeAhead;
-        }
-        else
-        {
-            mCslTimer.FireAt(mCslSampleTime + timeAfter);
-            mIsCslSampling = true;
-            winStart       = ot::TimerMicro::GetNow().GetValue();
-        }
-
-        winDuration = timeAhead + timeAfter;
+        uint32_t periodUs = mCslPeriod * kUsPerTenSymbols;
+        uint32_t winStart;
+        uint32_t winDuration;
+
+        mCslTimer.FireAt(mCslSampleTime + aTimeAfter);
+        mIsCslSampling = true;
+        winStart       = TimerMicro::GetNow().GetValue();
+        winDuration    = aTimeAhead + aTimeAfter;
         mCslSampleTime += periodUs;
 
         Get<Radio>().UpdateCslSampleTime(mCslSampleTime.GetValue());
-
-        // Schedule reception window for any state except RX - so that CSL RX Window has lower priority
-        // than scanning or RX after the data poll.
-        if (RadioSupportsReceiveTiming() && (mState != kStateDisabled) && (mState != kStateReceive))
-        {
-            IgnoreError(Get<Radio>().ReceiveAt(mCslChannel, winStart, winDuration));
-        }
-        else if (mState == kStateCslSample)
+        if (mState == kStateCslSample)
         {
             IgnoreError(Get<Radio>().Receive(mCslChannel));
         }
 
-        LogDebg("CSL window start %lu, duration %lu", ToUlong(winStart), ToUlong(winDuration));
+        LogCslWindow(winStart, winDuration);
+    }
+}
+
+void SubMac::HandleCslTimer(Timer &aTimer) { aTimer.Get<SubMac>().HandleCslTimer(); }
```

Comment:
How about moving the `HandleCslTimer()` method before the `HandleCslReceiveAt()` and `HandleCslReceiveOrSleep()` methods (around line ~136)? 

This way, I think it may help make the `git diff` smaller (since it is close to where it was defined before) also help someone reading the code. We would read `HandleCslTimer()` and see that it will use either `HandleCslReceiveAt()` or `HandleCslReceiveOrSleep()`, and they immediately follow. 😃